docusaurus/docs/iOS/components/message-layout-options-resolver.md

```diff
@@ -0,0 +1,154 @@
+---
+title: ChatMessageLayoutOptionsResolver
+---
+
+import ComponentsNote from '../common-content/components-note.md'
+import Properties from '../common-content/reference-docs/stream-chat-ui/chat-message-list/chat-message/chat-message-layout-options-properties.md'
+
+The `ChatMessageLayoutOptionsResolver` object is responsible for assigning layout options to a message in a specific position inside a list. Layout options are stored in the `ChatMessageLayoutOptions` struct type.
+
+Layout options are used by the message view to determine how the message should be rendered (ie. render the message with its reactions, message is leading a group of messages, ...).
+
+The SDK comes with a built-in resolver which follows the layout rules set by Stream Chat design, this can be customized by writing your own resolver.
+
+## Customization
+
+You can change how your messages are rendered by the message view component by selecting your own set of layout options. For instance, StreamChat groups messages by user and shows the avatar and user name at the end of the group. If in your application you want the avatar to be repeated for all messages (like Youtube does) you need to implement your own `ChatMessageLayoutOptionsResolver` class and register it.
+
+```swift
+import StreamChat
+import StreamChatUI
+import UIKit
+
+final class YTMessageLayoutOptionsResolver: ChatMessageLayoutOptionsResolver {
+    override func optionsForMessage(
+        at indexPath: IndexPath,
+        in channel: ChatChannel,
+        with messages: AnyRandomAccessCollection<ChatMessage>
+    ) -> ChatMessageLayoutOptions {
+        var options = super.optionsForMessage(at: indexPath, in: channel, with: messages)
+        
+        // Remove the message options that are not needed in our case
+        options.remove([
+            .flipped,
+            .bubble,
+            .timestamp,
+            .avatar,
+            .avatarSizePadding,
+            .authorName,
+            .threadInfo,
+            .reactions,
+            .onlyVisibleForYouIndicator,
+            .errorIndicator
+        ])
+        
+        // Always show the avatar, timestamp and author name for each message
+        options.insert([.avatar, .timestamp, .authorName])
+        
+        return options
+    }
+}
+```
+
+You can swap the built-in resolver with your own by setting `Components.default.channelContentView` to your own view type.
+
+```swift
+Components.default.messageLayoutOptionsResolver = YTMessageLayoutOptionsResolver()
+```
+
+<ComponentsNote />
+
+## ChatMessageLayoutOptions
+
+Describes the layout for a message based on its content and position in the message list. Views rendering a message should use this struct to determine how a message should be rendered.
+
+### Properties
+
+<Properties />
+
+## Examples
```

Comment:
It feels weird that the Examples of the customizations are not inside the "Customization" section 🤔

Reply:
atm I am providing specific examples in all important sections, looks easier to explain what something does